### DISCUSSION TITLE
Need to run only recognition inference. Inputs are image, detection result, batch size

### DISCUSSION THREAD

**Discussion by hchintada** [open-mmlab/mmocr#1867] [Q&A] [answered]
I need to run detection and recognition separately by manually passing the detection output as input to the recognition inference.

I can't use the 'det_rec' mode in the inference api scripts since here the detection result will come as one of the inputs. Facing difficulties trying to override `forward` or `__call__` methods to accept `detection_result` as an argument.

Any suggestions?

**Comment by gaotongxiao** [comment]
Why not use `TextRecInferencer`? https://mmocr.readthedocs.io/en/latest/user_guides/inference.html

**Reply by Insktall** [reply, marked as answer]
Modify the existing script: One option is to modify the existing script to accept the output of the detection model as input to the recognition model. Depending on the specific API you are using, this may involve modifying the forward or __call__ methods as you mentioned, or it may require modifying other parts of the code. This approach can be more challenging, especially if you are not familiar with the API codebase, but it allows you to take advantage of the existing functionality and avoid duplicating effort. Write your own script: Another option is to write your own inference script that performs separate detection and recognition steps. This approach can be more flexible and allow for more customization, but it also requires more effort. Depending on the specific models you are using, you may need to write custom code to preprocess the output of the detection model, such as cropping and resizing images, before passing them as input to the recognition model.Use a workflow management tool: If you are working with multiple models or stages in your pipeline, it may be helpful to use a workflow management tool like Apache Airflow or Kubeflow to manage the input and output of each step. This can help ensure that the correct outputs are passed as inputs to subsequent steps and can make it easier to scale and manage your pipeline.  

Ultimately, the best approach will depend on your specific use case and the specific tools and models you are using. If you are having trouble getting started, I would recommend consulting the documentation for the API you are using or reaching out to the community for support.
